catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java

```diff
@@ -584,17 +592,30 @@ private boolean validateFilePath(File resourceFilePath) throws IOException {
   }
 
   protected WebClient getWebClient(String uri, Map<String, Serializable> properties) {
-    WebClient client;
+    SecureCxfClientFactory<WebClient> factory;
     if (properties.get(USERNAME) != null && properties.get(PASSWORD) != null) {
-      client =
-          WebClient.create(
-              uri, (String) properties.get(USERNAME), (String) properties.get(PASSWORD), null);
+      factory =
+          clientFactoryFactory.getSecureCxfClientFactory(
+              uri,
+              WebClient.class,
+              null,
+              null,
+              false,
+              getFollowRedirects(),
+              null,
+              null,
+              (String) properties.get(USERNAME),
+              (String) properties.get(PASSWORD));
     } else {
-      client = WebClient.create(uri);
+      factory =
+          clientFactoryFactory.getSecureCxfClientFactory(
+              uri, WebClient.class, null, null, false, getFollowRedirects());
     }
-
-    WebClient.getConfig(client).getHttpConduit().getClient().setAutoRedirect(getFollowRedirects());
-    return client;
+    Serializable subject = properties.get(SecurityConstants.SECURITY_SUBJECT);
+    if (subject instanceof org.apache.shiro.subject.Subject) {
+      return factory.getClientForSubject((org.apache.shiro.subject.Subject) subject);
+    }
```

Comment:
✏️ This is redundant with logic in the `retrieveHttpProduct ` method. That logic should be removed. https://github.com/codice/ddf/blob/aa2174df245707f5f0ea2186c4897359b16c0db7/catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java#L364-L369